components/eamxx/src/share/field/field_utils.hpp

```diff
@@ -43,6 +43,41 @@ void randomize (const Field& f, Engine& engine, PDF&& pdf)
   impl::randomize<ST>(f,engine,pdf);
 }
 
+// Compute a random perturbation of a field for all view entries
+// field_view(:, k) when level_mask(k)==true. The field
+// must have level midpoint tag as last dimension, and level mask
+// should have size of last field dimension.
+template<typename Engine, typename PDF, typename MaskType>
+void perturb (const Field& f, Engine& engine, PDF&& pdf,
+	      const MaskType& level_mask)
```

Comment:
I chose to use a level mask. I could make it even more generic and make a mask for every entry. @PeterCaldwell can you envision any use case where we want to perturb based on anything but level index?

Reply:
> can you envision any use case where we want to perturb based on anything but level index?
No. Maybe someday someone will come up with a use case, but let's let them add that functionality if/when they do in order to make our code simpler now.